src/child/actions/window.js

```diff
@@ -99,40 +157,40 @@ function resizeCompact() {
                 compactWindowWidth,
                 compactWindowHeight,
                 'top-right',
-                () => resolve(dispatch(compact())),
+                () => resolve(dispatch(resizeCompactSuccess(isMaximised))),
                 () => reject(dispatch(resizeError()))
             );
         });
     };
 }
 
-function resizeDefault() {
+function resizePrevious(previousExpandedDimensions, previousMaximisedState) {
     return dispatch => {
         dispatch(resizing());
-        const [defaultWindowWidth, defaultWindowHeight] = configService.getDefaultWindowDimensions();
+        const [previousWindowWidth, previousWindowHeight] = previousExpandedDimensions;
 
         return new Promise((resolve, reject) => {
             currentWindowService.resizeTo(
-                defaultWindowWidth,
-                defaultWindowHeight,
+                previousWindowWidth,
+                previousWindowHeight,
                 'top-right',
-                () => resolve(dispatch(expand())),
+                () => resolve(dispatch(resizePreviousSuccess(previousMaximisedState))),
                 () => reject(dispatch(resizeError()))
             );
         });
     };
 }
 
-export function resizeToCompact() {
+export function resizeToCompact(isMaximised) {
```

Comment:
I'm not sure I'd add the param here - if the value is just coming from the state, then why not get the state in the thunk, using `getState()`?

> The inner function receives the store methods `dispatch` and `getState` as parameters.

(https://github.com/gaearon/redux-thunk#motivation)